File: app/src/main/java/org/tasks/receivers/MyPackageReplacedReceiver.java
package org.tasks.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import timber.log.Timber;

public class MyPackageReplacedReceiver extends BroadcastReceiver {

  @Override
  public void onReceive(Context context, Intent intent) {
    if (!Intent.ACTION_MY_PACKAGE_REPLACED.equals(intent.getAction())) {
      return;
    }

    Timber.d("onReceive(context, %s)", intent);
  }
}

File: app/src/main/java/org/tasks/receivers/MyPackageReplacedReceiver.kt
package org.tasks.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.tasks.location.RegisterGeofencesWork
import timber.log.Timber

class MyPackageReplacedReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action != Intent.ACTION_MY_PACKAGE_REPLACED) return
        Timber.d("Package replaced, re-registering geofences")
        RegisterGeofencesWork.enqueue(context)
    }
}
